### PR TITLE
gqltest: migrate structural search and And/Or queries

### PR DESCRIPTION
**Pull request by unknwon**
Migrates structural search and And/Or queries to the `dev/gqltest` test suite.

### After merge

- [x] Help @rvantonder use this new test suite locally for regular development.
- [ ] Remove `internal/cmd/search-integration-tester`.

### CODE REVIEW

File: dev/gqltest/search_test.go [hunk 1]
```diff
@@ -457,4 +457,239 @@ func TestSearch(t *testing.T) {
...
+				wantAlert: &gqltestutil.SearchAlert{
+					Title:       "Unable To Process Query",
+					Description: "Error parsing regexp: missing closing ): `(respObj.Size`",
+				},
+			},
+			{
+				// TODO: No alert even for the original query
+				// 	repo:^github\.com/rvantonderp/DirectXMan12-k8s-prometheus-adapter$@4b5788e file:^README\.md (bar and (foo or x\) ()) patterntype:literal
+				name:       `Confusing grouping raises alert`,
+				query:      `repo:^github\.com/sgtest/go-diff file:^README\.md (bar and (foo or x\) ()) patterntype:literal`,
```
Comment (unknwon) [inline dev/gqltest/search_test.go:624]: @rvantonder: Could you help with this case? My tests show no alert created for this query, wondering if it's still the case?

**Comment by rvantonder** [comment]
Will look into this Tue Sept 15!

File: dev/gqltest/search_test.go (hunk 1 above)
Comment (rvantonder) [reply dev/gqltest/search_test.go:624]: This should just be No results (`zeroResult: true` is fine). I think I changed the behavior at some point.

File: dev/gqltest/search_test.go [hunk 2]
```diff
@@ -457,4 +457,241 @@ func TestSearch(t *testing.T) {
...
+				zeroResult: true,
+				wantAlert: &gqltestutil.SearchAlert{
+					Title:       "Unable To Process Query",
+					Description: "Error parsing regexp: missing closing ): `(respObj.Size`",
+				},
+			},
+			{
+				// TODO: No alert even for the original query
+				// 	repo:^github\.com/rvantonderp/DirectXMan12-k8s-prometheus-adapter$@4b5788e file:^README\.md (bar and (foo or x\) ()) patterntype:literal
+				name:       `Confusing grouping raises alert`,
```
Comment (rvantonder) [inline dev/gqltest/search_test.go:623]:
```suggestion
				name:       `No result for confusing grouping`,
```

**Review by rvantonder — APPROVED** [review]
This is so great. If the current tests pass, just go ahead and merge.

I tried to run things locally by following the README, but couldn't quite get it working. Will schedule a call to run through things with you and can then audit test output if needed.

**Comment by unknwon** [comment]
Merging since `master-dry-run` PR passed CI: https://github.com/sourcegraph/sourcegraph/pull/13905

**Pull request merged**